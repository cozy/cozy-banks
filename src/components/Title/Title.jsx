import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { MainTitle } from 'cozy-ui/react/Text'
import styles from './Title.styl'

class Title extends React.PureComponent {
  render() {
    const { children, color, className } = this.props

    return (
      <MainTitle
        tag="h1"
        ellipsis={true}
        className={cx(styles.Title, styles[`TitleColor_${color}`], className)}
      >
        {children}
      </MainTitle>
    )
  }
}

Title.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf(['default', 'primary'])
}

Title.defaultProps = {
  color: 'default'
}

export default Title
