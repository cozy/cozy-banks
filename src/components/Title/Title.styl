@require 'settings/breakpoints'

.Title
    line-height 2.5rem
    margin 0 auto

.TitleColor_default
    color var(--black) !important

.TitleColor_primary
    color var(--primary-contrast-text) !important

+small-screen()
    .Title
        font-size 1.125rem
        line-height 1.5rem
        margin-bottom 0
