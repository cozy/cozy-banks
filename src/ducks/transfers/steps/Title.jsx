import React from 'react'
import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
import Typography from 'cozy-ui/transpiled/react/Typography'

const Title = ({ children }) => {
  const { isMobile } = useBreakpoints()
  return (
    <Typography
      variant="h4"
      className={'u-mb-1 ' + (isMobile ? 'u-ta-center' : '')}
    >
      {children}
    </Typography>
  )
}

export default React.memo(Title)
