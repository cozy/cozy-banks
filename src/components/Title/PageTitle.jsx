import React from 'react'
import PropTypes from 'prop-types'
import Title from 'components/Title/Title'
import { BarCenter } from 'components/Bar'
import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'

/**
 * Renders a title in the bar
 *
 * Use it only if the page structure between mobile and desktop is very different
 * and you cannot rely on PageTitle
 */
export const BarTitle = ({ children, className }) => {
  return (
    <BarCenter>
      <Title className={className}>{children}</Title>
    </BarCenter>
  )
}

/**
 * Render a page title responsively
 *
 * Will display the title inside the bar on mobile and directly
 * in the page (in a Title style) otherwise
 */
const PageTitle = ({ children, className }) => {
  const { isMobile } = useBreakpoints()
  return isMobile ? (
    <BarTitle className={className}>{children}</BarTitle>
  ) : (
    <Title className={className}>{children}</Title>
  )
}

PageTitle.propTypes = {
  children: PropTypes.node
}

export default PageTitle
