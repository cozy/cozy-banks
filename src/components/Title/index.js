import Title from 'components/Title/Title'
import PageTitle from 'components/Title/PageTitle'

export { PageTitle, Title }

export default Title
