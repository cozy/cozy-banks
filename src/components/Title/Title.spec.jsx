import React from 'react'
import { shallow } from 'enzyme'
import Title from './Title'
import CozyTheme from 'cozy-ui/transpiled/react/providers/CozyTheme'

describe('Title', () => {
  it(`should display children`, () => {
    expect(shallow(<Title>content</Title>).getElement()).toMatchSnapshot()
  })

  it(`should extend className`, () => {
    expect(
      shallow(<Title className="noPaddingBottom">content</Title>).getElement()
    ).toMatchSnapshot()
  })

  it(`should handle theme`, () => {
    expect(
      shallow(
        <CozyTheme variant="inverted">
          <Title>content</Title>
        </CozyTheme>
      ).getElement()
    ).toMatchSnapshot()
  })
})
