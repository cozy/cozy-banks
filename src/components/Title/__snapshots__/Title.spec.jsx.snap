// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should display children 1`] = `
<ForwardRef
  className="Title"
  variant="h4"
>
  content
</ForwardRef>
`;

exports[`Title should extend className 1`] = `
<ForwardRef
  className="Title noPaddingBottom"
  variant="h4"
>
  content
</ForwardRef>
`;

exports[`Title should handle theme 1`] = `
<Context.Provider
  value="inverted"
>
  <MuiCozyTheme
    variant="inverted"
  >
    <div
      className="palette__CozyTheme--inverted___UL0fq"
    >
      <Title>
        content
      </Title>
    </div>
  </MuiCozyTheme>
</Context.Provider>
`;
